spec/Dynamic-Validator.md

```diff
@@ -0,0 +1,134 @@
+                        |          +--------+    |       |
+                       (6)                      (6)      |
+                        |                        |       |
++----------+ -(1)--> +-----------+ -(3)--> +-----------+ |
+| Eligible |         | Candidate |         | Validator | |
++----------+ <--(2)- +-----------+ <--(4)- +-----------+ |
+      ^                 ^                        |       |
+      |                 |      +--------+ <--(5)-/       |
+      |                 \-(7)- | Jailed | -(6)-----------/
+      \-------------------(9)- +--------+
+                                 ^     |
+                                 |     |
+                                 \-(8)-/
+```
+1. Send *SELF_NOMINATE*
+2. No *SELF_NOMINATE* while **NOMINATE_EXPIRATION** terms
+3. Elected
+4. End of term and the validator worked
+5. End of term and the validator didn't work
+6. Double Vote detected
+7. Send *SELF_NOMINATE* after **CUSTODY_PERIOD**
+8. Send *SELF_NOMINATE* before **CUSTODY_PERIOD**
+9. No *SELF_NOMINATE* during **RELEASE_PERIOD**
+
+## Term
+The term is a period that one elected validator set works.
+It is for almost one hour.
+The block that's generation hour is different from the parent block's is the last block of a term.
+CodeChain elects new validator set after all reward of the block is given.
+
+## Nomination
+Any account, that is not banned, can nominate itself.
+The account becomes a candidate when the sum of the deposit is more than **MIN_DEPOSIT**.
+The nomination expires after **NOMINATION_EXPIRATION**; the account who wants to remain a candidate must nominate itself before the previous nomination expires.
+The deposit reverts to the account when it becomes an eligible account.
+
+### Minimum Deposit
+TBD
+
+## Delegation
+The stakeholders have the right to choose validators as much as their shares.
+It's called a delegation.
+And the stakeholders who have been delegated called delegators.
+The delegation is valid only when the delegatee is not an eligible or not banned.
+The delegated stakes are returned when the account becomes an eligible or banned account.
+
+## Election
+The process that elects validators of a term is called the election.
+The election selects validators as the following rule.
+
+1. TBD
+
+## Voting Power
+Each elected validators has different voting power.
+The voting power is based on the delegation that the validator has received at the election.
+The block is valid only if the sum of voting power is more than 2/3 of the total delegations that elected validators received.
+
+## Validator Reward
+The block proposer gets the express fee of the blocks at the end of a term.
+Validators can get the reward after **WITHDRAW_DELAY** terms.
+However, the proposers cannot get all reward if he is not loyal to their duty.
+The reward is decreased according to the rate of the blocks the validator misses to sign.
+TBD: The rate of decreasing.
```

Comment:
The block proposer gets the express fee of the blocks at the end of a term.
Validators can get the reward after **WITHDRAW_DELAY** terms; however, the proposers cannot get all the reward if they are not loyal to their duty.
The reward is decreased according to the rate of the blocks the validator misses to sign.
TBD: The rate of decreasing.